guides/source/active_record_encryption.md

```diff
@@ -51,71 +59,174 @@ config.active_record.encryption.key_derivation_salt = ENV["ACTIVE_RECORD_ENCRYPT
 
 NOTE: These generated values are 32 bytes in length. If you generate these yourself, the minimum lengths you should use are 12 bytes for the primary key (this will be used to derive the AES 32 bytes key) and 20 bytes for the salt.
 
-### Declaration of Encrypted Attributes
+Once the keys are generated and stored, you can start using Active Record Encryption by declaring attributes to be encrypted.
 
-Encryptable attributes are defined at the model level. These are regular Active Record attributes backed by a column with the same name.
+### Declare Encrypted Attributes
+
+The [`encrypts` method](https://api.rubyonrails.org/classes/ActiveRecord/Encryption/EncryptableRecord.html#method-i-encrypts) defines attribute to be encrypted at the model level. These are regular Active Record attributes backed by a column with the same name.
 
 ```ruby
 class Article < ApplicationRecord
   encrypts :title
 end
-````
+```
 
-The library will transparently encrypt these attributes before saving them in the database and will decrypt them upon retrieval:
+Active Record Encryption library will transparently encrypt these attributes before saving them to the database and will decrypt them upon retrieval. For example,
 
 ```ruby
 article = Article.create title: "Encrypt it all!"
 article.title # => "Encrypt it all!"
 ```
 
-But, under the hood, the executed SQL looks like this:
+However, in the Rails console, the executed SQL looks like this:
 
 ```sql
 INSERT INTO `articles` (`title`) VALUES ('{\"p\":\"n7J0/ol+a7DRMeaE\",\"h\":{\"iv\":\"DXZMDWUKfp3bg/Yu\",\"at\":\"X1/YjMHbHD4talgF9dt61A==\"}}')
 ```
 
-#### Important: About Storage and Column Size
+### Querying Encrypted Data: Deterministic vs. Non-deterministic Encryption
+
+By default, Active Record Encryption is non-deterministic, which means that encrypting the same value with the same key twice will result in *different* encrypted values (aka ciphertexts). The non-deterministic approach improves security by making crypto-analysis of ciphertexts harder.  However, it makes querying the database impossible.
+
+If you need to able to query the encrypted `email` field on the `Author` model below, you can use deterministic encryption:
```

Comment:
What do you think about using the same model from the example above to follow through in examples below? Or updating the model above to match the model below?